File: examples/example_gripper.java
package org.sdk.lebai.examples;

import org.sdk.lebai.Gripper;
import org.sdk.lebai.Loader;

public final class example_gripper {
  public static void main(String[] args) {
    if (args.length != 1) {
      System.err.println("Usage: example_gripper <serial_port>");
      System.err.println("Example (Windows): example_gripper COM3");
      System.err.println("Example (Linux): example_gripper /dev/ttyUSB0");
      return;
    }

    Loader.loadNativeLibraries();
    Gripper gripper = new Gripper(args[0]);
    gripper.set_position(30);
    gripper.set_position(70);
    gripper.do_calibration();
    System.out.println("Calibrated: " + gripper.is_calibrated());
  }

  private example_gripper() {}
}

File: examples/example_gripper.cs
using System;
using lebai.gripper;

namespace lebai.app
{
    class ExampleGripper
    {
        static int Main(string[] args)
        {
            if (args.Length != 1)
            {
                Console.Error.WriteLine($"Usage: {AppDomain.CurrentDomain.FriendlyName} <serial_port>");
                Console.Error.WriteLine($"Example (Windows): {AppDomain.CurrentDomain.FriendlyName} COM3");
                Console.Error.WriteLine($"Example (Linux): {AppDomain.CurrentDomain.FriendlyName} /dev/ttyUSB0");
                return 0;
            }

            try
            {
                using (var gripper = new Gripper(args[0]))
                {
                    Console.WriteLine("Gripper connected.");
                    gripper.set_position(30);
                    gripper.set_position(70);
                    gripper.do_calibration();
                    Console.WriteLine($"Calibrated: {gripper.is_calibrated()}");
                }
                return 0;
            }
            catch (Exception ex)
            {
                Console.Error.WriteLine($"Error: {ex.Message}");
                return 1;
            }
        }
    }
}
